util/topsql/tracecpu/profile.go

```diff
@@ -15,34 +15,26 @@
 package tracecpu
 
 import (
-	"bytes"
 	"context"
-	"errors"
-	"fmt"
-	"io"
-	"net/http"
 	"runtime/pprof"
-	"strconv"
 	"sync"
-	"sync/atomic"
 	"time"
 
 	"github.com/google/pprof/profile"
-	"github.com/pingcap/tidb/sessionctx/variable"
 	"github.com/pingcap/tidb/util"
+	"github.com/pingcap/tidb/util/cpuprofile"
 	"github.com/pingcap/tidb/util/hack"
 	"github.com/pingcap/tidb/util/logutil"
 	"go.uber.org/zap"
 )
 
 const (
-	labelSQL        = "sql"
 	labelSQLDigest  = "sql_digest"
 	labelPlanDigest = "plan_digest"
 )
 
-// GlobalSQLCPUProfiler is the global SQL stats profiler.
-var GlobalSQLCPUProfiler = newSQLCPUProfiler()
+// GlobalSQLCPUCollector is the global SQL CPU stats collector.
+var GlobalSQLCPUCollector = newSQLCPUCollector()
```

Comment:
Consider removing this global variable. I think it can be a field of the reporter.

Reply:
great, done.